File: mobile/src/main/java/treehou/se/habit/ui/servers/create/scan/ScanServersComponent.java
package treehou.se.habit.ui.servers.create.scan;


import dagger.Subcomponent;
import treehou.se.habit.module.ActivityScope;
import treehou.se.habit.module.FragmentComponent;
import treehou.se.habit.module.FragmentComponentBuilder;
import treehou.se.habit.ui.servers.create.custom.ScanServersModule;

@ActivityScope
@Subcomponent(
        modules = ScanServersModule.class
)
public interface ScanServersComponent extends FragmentComponent<ScanServersFragment> {

    @Subcomponent.Builder
    interface Builder extends FragmentComponentBuilder<ScanServersModule, ScanServersComponent> {
    }
}

File: mobile/src/main/java/treehou/se/habit/dagger/fragment/ScanServersComponent.kt
package treehou.se.habit.dagger.fragment


import dagger.Subcomponent
import treehou.se.habit.dagger.FragmentComponent
import treehou.se.habit.dagger.FragmentComponentBuilder
import treehou.se.habit.dagger.scopes.ActivityScope
import treehou.se.habit.ui.servers.create.scan.ScanServersFragment

@ActivityScope
@Subcomponent(modules = arrayOf(ScanServersModule::class))
interface ScanServersComponent : FragmentComponent<ScanServersFragment> {

    @Subcomponent.Builder
    interface Builder : FragmentComponentBuilder<ScanServersModule, ScanServersComponent>
}